libs/sdk-core/src/greenlight/node_api.rs

```diff
@@ -1452,13 +1458,40 @@ async fn pull_transactions(
         .map(TryInto::try_into)
         .collect();
 
+    let outbound_transactions: NodeResult<Vec<Payment>> =
+        htlc_expiry(outbound_transactions.unwrap(), htlc_list);
+
     let mut transactions: Vec<Payment> = Vec::new();
     transactions.extend(received_transactions?);
     transactions.extend(outbound_transactions?);
 
     Ok(transactions)
 }
 
+fn htlc_expiry(payments: Vec<Payment>, htlc_list: Vec<Htlc>) -> NodeResult<Vec<Payment>> {
+    let mut pending_payments: Vec<Payment> = Vec::new();
```

Comment:
From what I understand in `pull_transactions` the payments could be pending or completed, maybe rename to something more general like `payments_res`. I assume if the payment is completed there will be no matching HTLC, so no need to filter.